it/controllers/pensions/incomeFromPensions/PensionAmountControllerISpec.scala

```diff
@@ -0,0 +1,509 @@
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    val expectedTitle = "How much pension did your client get paid?"
+    val expectedHeading = "How much pension did your client get paid?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY)))
+
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      import Selectors._
+      import user.commonExpectedResults._
+
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        "render How much pension did you get paid with no prefilled value for tax paid and total tax" which {
+          lazy val pensionIncomeModel = UkPensionIncomeViewModel(
+            amount = None, taxPaid = None)
+
+          lazy val result: WSResponse = {
+            dropPensionsDB()
+            authoriseAgentOrIndividual(user.isAgent)
+            insertCyaData(pensionsUserDataWithIncomeFromPensions(
+              anIncomeFromPensionsViewModel.copy(uKPensionIncomesQuestion = Some(true), uKPensionIncomes = Seq(pensionIncomeModel))), aUserRequest)
+            urlGet(fullUrl(pensionAmountUrl(taxYearEOY, index)), user.isWelsh, follow = false,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          textOnPageCheck(totalTax, labelIndex(2))
+          textOnPageCheck(poundPrefixText, poundPrefixSelector(2), "for amount 1")
+          textOnPageCheck(hintText, amount1hintTextSelector, "for amount 1")
+          textOnPageCheck(taxPaid, labelIndex(3))
+          textOnPageCheck(hintText, amount2hintTextSelector, "for amount 2")
+          inputFieldValueCheck(amount1InputName, amount1inputSelector, "")
+          inputFieldValueCheck(amount2InputName, amount2inputSelector, "")
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(pensionAmountUrl(taxYearEOY, index), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render How much pension did you get paid with no prefilled value for tax paid when total paid is None" which {
```

Comment:
think the test name here is switched, should be prefilled total paid, and taxPaid is None